example/src/Booking/Api.hs

```diff
@@ -0,0 +1,23 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE TypeOperators         #-}
+
+module Booking.Api where
+
+import           Booking.Types
+import           Data.Set      (Set)
+import           Data.Text     (Text)
+import           Servant
+
+
+type BookingAPI = "hotelbooking" :>
+       -- ^ 1) Return a set of availables rooms based on the CheckInOut dates
+  (    "getAvailableRooms" :> ReqBody '[JSON] CheckInOut :> Get '[JSON] (Set Room)
+       -- ^ 2) Return a boolean if the given RoomId is available
+  :<|> "isRoomAvailable" :> ReqBody '[JSON] Text :> Get '[JSON] Bool
```

Comment:
Here we do not need a JSON... it would be nicer to have just a `Capture` to send the RoomId in the request something like
`isRoomAvailable/104`
Check the documentation on captures:
 https://haskell-servant.readthedocs.io/en/stable/tutorial/Server.html#a-first-example